### DISCUSSION TITLE
Irregular electrode shape shown in gmsh visualization results

### DISCUSSION THREAD

**Discussion by liangsihan** [simnibs/simnibs#354] [9. Results and analysis]
Hi all,
I noticed that the electrode shape visualized in gmsh is somewhat irregular, even though I set the electrode shape as a ellipse with diameter 5cm * 5cm, so I expected it to be a circular shape. I attached the screenshot below. May I ask what might cause this problem, and is it something to worry about? 
<img width="443" alt="Screenshot 2024-06-03 at 11 34 36 PM" src="https://github.com/simnibs/simnibs/assets/86599135/8b2390cb-88ee-4971-9d85-ba71db804b48">
<img width="394" alt="Screenshot 2024-06-03 at 11 35 06 PM" src="https://github.com/simnibs/simnibs/assets/86599135/a9988682-ff2e-49ac-9efb-9284576f72fd">

**Comment by diedrim** [comment]
These slight deviations from the expected shape are normal. The electrode placement on the head mesh includes a shape quality control of the mesh tetrahedra. In order to ensure that no flat or pointed tetrahedra distort the field simulations, the electrode nodes are placed to approximate the desired shape as good as possible. Slight deformations will not influence the result of the simulation.
